pkg/apis/kubeflow/v1alpha2/types.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 The Kubeflow Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type MPIJob struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              MPIJobSpec `json:"spec,omitempty"`
+	Status            JobStatus  `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type MPIJobList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []MPIJob `json:"items"`
+}
+
+type ProcessingSpec struct {
+	// Specifies the desired number of processing units the MPIJob should run on.
+	// Mutually exclusive with `ReplicaSpec.Replicas` in `MPIReplicaSpecs`.
+	// +optional
+	Units *int32 `json:"units,omitempty"`
+
+	// The maximum number of processing units available per node.
+	// Note that this will be ignored if the processing resources are explicitly
+	// specified in the MPIJob pod spec.
+	// +optional
+	UnitsPerNode *int32 `json:"unitsPerNode,omitempty"`
+
+	// The processing resource type, e.g. 'nvidia.com/gpu' or 'cpu'.
+	// Defaults to 'nvidia.com/gpu'
+	// +optional
+	ResourceType string `json:"resourceType,omitempty"`
```

Comment:
Would be good to have constants for allowed ResourceTypes

Reply:
And I would suggest using [`corev1.ResourceName`](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L4162)  type instead of string.

Reply:
Thanks! Done.